text/0113-virtual-script-container.md

```diff
@@ -0,0 +1,216 @@
+
+### Globals
+
+Some scripts define globals which are meant to be used by the rest of the application. Let's take this example of Google Analytics:
+
+```html
+<!-- Google Analytics -->
+<script>
+    window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
+    ga('create', 'UA-XXXXX-Y', 'auto');
+    ga('send', 'pageview');
+</script>
+<script async src='https://www.google-analytics.com/analytics.js'></script>
+<!-- End Google Analytics -->
+```
+
+If we evaluate the above snippets with `<virtual-script>` instead, then `ga` global value will only be available into the sandbox. So `ga` will not be usable by anyone else in the entire application. We can however, allow for globals to be exposed into the main window. We call this global _exporting_. 
+
+Conversely, if a script running in `virtual-script` needs access to a global variable from the main window, we can _import_ this global. 
+
+A syntax like the following will be provided:
+
+```html
+<virtual-script exported-globals="foo"></virtual-script>
+<virtual-script imported-globals="bar"></virtual-script>
+```
+
+- `virtual-script` container would not have access to globals defined by other scripts in the outer window unless they were imported using `imported-globals`.
+
+- Globals defined in `exported-globals` will be redefined in any `virtual-script` container.
+- Exported globals will also be available to any Locker-created sandboxes
+- Globals defined in Oasis using `imported-globals` will be redefined only in the `virtual-script` container they are imported to.
+
+Note: these global values are controlled side-channels between the main window and the sandbox, and should be used with caution. For example, creating a component that depends on `ga` to be defined, is an anti-pattern.
+
+### Supported Attributes
+
+`crossorigin`: indicates that the browser should provide more information to window.onerror when handling CORS errors
+
+`defer`: indicates that the script should be executed after the document has parsed (has no effect if `src` is absent)
+
+`integrity`: enables the browser to verify that a fetched resource is delivered without unexpected manipulation
+
+`nomodule`: the script should not be executed in browsers that support ES2015 modules
+
+`nonce`: user provided nonce values for CSP will be supported
+
+`referrerpolicy`: indicates which referrer to send when fetching the script
+
+`src`: the address of the resource
+
+**async**
+
+TBD describe asynchronous nature of `virtual-script`.
+
+### Unsupported Attributes
+
+`type`: all `virtual-script` resources will be `text/javascript`. `type="module"` will not be supported.
+
+`charset`: this attribute is deprecated
+
+`language`: this attribute is deprecated
+
+`nonce`: has special semantics since it is a global attribute and browsers do special behavior with this attribute when a CSP policy is applied. We will have to treat this attribute specially.
```

Comment:
lets chat about this one

Reply:
I put some more information in this issue here: https://github.com/salesforce/oasis/issues/8